### ISSUE TITLE
Resolve standard options and verbosity

### ISSUE THREAD

**Issue by meshula** [AcademySoftwareFoundation/openexr#884]
From: https://github.com/AcademySoftwareFoundation/openexr/issues/881

- Please don't write messages unless it's an error. By default, I'm seeing:

```sh
 Configure ILMBASE Version: 2.5.3 Lib API: 25.0.2
 -- WARNING pkg-config generation disabled
 Configure OpenEXR Version: 2.5.3 Lib API: 25.0.2
```

This happens on every single configuration. That quickly adds up when every project does it.

- It would be great if all options are prefixed OPENEXR_ consistently

- Don't add CDashBoard targets unconditionally when built as a submodule (this can be worked around by setting BUILD_TESTING to off -- would be nice to have it off by default for submodules though.)

- if you're not the top-level project don't set global options. ~An easy check is to compare the CMAKE_CURRENT_SOURCE_DIR with CMAKE_SOURCE_DIR and not turn on global options if OpenEXR is a submodule.~

Note, rather than checking source directory, OpenEXR already has this code in place:

```cmake
if(NOT "${CMAKE_PROJECT_NAME}" STREQUAL "${PROJECT_NAME}")
  set(OPENEXR_IS_SUBPROJECT ON)
  message(NOTICE "OpenEXR is configuring as a cmake sub project")
endif()
```

reference:

https://github.com/AcademySoftwareFoundation/openexr/issues/881
https://github.com/AcademySoftwareFoundation/openexr/issues/871
https://github.com/AcademySoftwareFoundation/openexr/issues/833
https://github.com/AcademySoftwareFoundation/openexr/issues/748

**Comment by cary-ilm** [comment]
For the first point regarding errors, see #899.

**Comment by cary-ilm** [comment]
These should be addressed in the v3.0.0 release, so closing for now, but feel free to reopen if you encounter further problems.
